code/ZAS/Fire.dm

```diff
@@ -446,8 +446,7 @@ datum/gas_mixture/proc/calculate_firelevel(var/turf/T)
 	//Get heat transfer coefficients for clothing.
 
 	for(var/obj/item/clothing/C in src)
-		if(l_hand == C || r_hand == C)
-			continue
+		if(held_items.Find(C)) continue
```

Comment:
Focusing on encapsulation and usability, I suggest you make this a proc instead.
Something like is_in_hands(obj/item)